### ISSUE TITLE
discrepancies between missing aggregation and terms aggregation with missing parameter

### ISSUE THREAD

**Issue by asldevi** [elastic/elasticsearch#18675]
**Elasticsearch version**: 2.1.1

**JVM version**: 1.8.0_25

**OS version**: Mac OS X EI Capitan

**Steps to reproduce**:

*Missing aggregation*

```
GET /.../_search?size=0
{
    "aggs" : {
        "wo_exp_level": {"missing": {"field": "experience_level"}},
        "exp_level" : { 
            "terms" : { "field" : "experience_level"}}
    }
}
```
returns 
```
 "exp_level": {
         "doc_count_error_upper_bound": 0,
         "sum_other_doc_count": 0,
         "buckets": [
            {
               "key": "Mid-Level",
               "doc_count": 1466
            },
            {
               "key": "Senior",
               "doc_count": 1164
            },
            {
               "key": "Lead",
               "doc_count": 854
            },
            {
               "key": "Junior",
               "doc_count": 824
            }
         ]
      },
      "wo_exp_level": {
         "doc_count": 65692
	}
```

*Terms aggregation with missing parameter*

```
GET /.../_search?size=0
{
    "aggs" : {
        "exp_level" : { 
            "terms" : { "field" : "experience_level", "missing": "notset"}}
    }
}
```
gives 
```
"exp_level": {
         "doc_count_error_upper_bound": 0,
         "sum_other_doc_count": 0,
         "buckets": [
            {
               "key": "notset",
               "doc_count": 65574
            },
            {
               "key": "Mid-Level",
               "doc_count": 1463
            },
            {
               "key": "Senior",
               "doc_count": 1274
            },
            {
               "key": "Lead",
               "doc_count": 863
            },
            {
               "key": "Junior",
               "doc_count": 826
            }
         ]
      }
```

I expect the value in the bucket `notset` (65574) to be same as `doc_count` in the missing aggregation (65692), which is `wo_exp_level`. But they're not. Also the counts in other buckets also change with and without the `missing` parameter.

**Comment by jpountz** [comment]
I am wondering that this might have been addressed by #15746. Could you check whether you can reproduce the bug with a more recent version of elasticsearch?

**Comment by asldevi** [comment]
I checked it with `elastic 2.3.3` and that works perfect. Thank you so much.
Team elastic is simply awesome 💯
